File: Abubakar Omolaja/Day 4/productofarrayexceptself.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        if len(nums)==0:
            return []
        n = len(nums)
        result = [0 for i in range(len(nums))]
        result[0]=1
        rp = 1
        for i in range(1,n,1):
            rp = rp *nums[i-1]
            result[i] = rp
        rp = 1
        for i in range(n-2,-1,-1):
            rp = rp * nums[i+1]
            result[i] = result[i]*rp
        return result

File: Mr-Caesar/Day-4/product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
 var productExceptSelf = function(nums) {
  let result = new Array(nums.length).fill(1)
  let postFix = 1
  
  for(let i=1; i< nums.length; i++ ) {
      result[i] = result[i-1] * nums[i-1]
  }
  
  for(let j=nums.length - 1; j >= 0; j--) {
      result[j] = result[j] * postFix
      console.log(result[j])
      postFix = postFix * nums[j]
  }
  
  return result
};